source/iterators.tex

```diff
@@ -1191,43 +1197,51 @@
 and \tcode{\placeholder{ITER_CONCEPT}(I)} denotes \tcode{random_access_iterator_tag}.
 \end{example}
 
-\rSec3[iterator.concept.readable]{Concept \cname{readable}}
+\rSec3[iterator.concept.readable]{Concept \cname{indirectly_readable}}
 
 \pnum
-Types that are readable by applying \tcode{operator*}
-model the \libconcept{readable} concept, including
+Types that are indirectly readable by applying \tcode{operator*}
+model the \libconcept{indirectly_readable} concept, including
 pointers, smart pointers, and iterators.
 
 \begin{codeblock}
 template<class In>
-  concept @\deflibconcept{readable}@ =
-    requires {
+  concept @\defexposconcept{indirectly-readable-impl}@ =
+    requires(const In in) {
       typename iter_value_t<In>;
       typename iter_reference_t<In>;
       typename iter_rvalue_reference_t<In>;
+      { *in } -> same_as<iter_reference_t<In>>;
+      { iter_move(in) } -> same_as<iter_rvalue_reference_t<In>>;
     } &&
     common_reference_with<iter_reference_t<In>&&, iter_value_t<In>&> &&
     common_reference_with<iter_reference_t<In>&&, iter_rvalue_reference_t<In>&&> &&
     common_reference_with<iter_rvalue_reference_t<In>&&, const iter_value_t<In>&>;
 \end{codeblock}
 
+\begin{codeblock}
+template<class In>
+  concept @\deflibconcept{indirectly_readable}@ =
+    @\placeholdernc{indirectly-readable-impl}@<remove_cvref_t<In>>;
+\end{codeblock}
+
 \pnum
-Given a value \tcode{i} of type \tcode{I}, \tcode{I} models \libconcept{readable}
+Given a value \tcode{i} of type \tcode{I}, \tcode{I} models \libconcept{indirectly_readable}
 only if the expression \tcode{*i} is equality-preserving.
 \begin{note}
 The expression \tcode{*i} is indirectly required to be valid via the
-exposition-only \placeholder{dereferenceable} concept\iref{iterator.synopsis}.
+exposition-only \tcode{\placeholder{dereferenceable}} concept\iref{iterator.synopsis}.
```

Comment:
This should be `\exposconcept`

Reply:
We use `\placeholder` throughout this clause for this purpose:

```
\pnum
The definitions in this subclause make use of the following
exposition-only concepts:

\begin{codeblock}
template<class I>
concept @\placeholder{cpp17-iterator}@ =
  copyable<I> && requires(I i) {
...
```

```
Otherwise, if \tcode{I} satisfies the exposition-only concept
\tcode{\placeholder{cpp17-input-iterator}},
```

(etc). I think this cleanup should be done as a single pass across all the exposition-only concepts, if we really think this level of detail in our markup is worthwhile. (Why are we singling out concepts for this treatment?)

Reply:
> Why are we singling out concepts for this treatment?

Because we have a special "index of concept names" that, ideally, should point to every mention of a concept. That's why I made `\deflibconcept` and `\libconcept` in the first place.
I've created #3494 for the global cleanup.